Yafc.Model/Data/DataUtils.cs

```diff
@@ -461,57 +463,105 @@ public class FavoritesComparer<T>(Project project, IComparer<T> def) : IComparer
+        /// Values are accepted in any format that YAFC can display, and consist of:<br/>
+        /// * A floating point number<br/>
+        /// * An optional SI prefix from μukMGT, case sensitive only for u and μ. (For historical reasons, m and M are both mega-; the milli- prefix is unused and unavailable.)<br/>
+        /// * An optional W, J, s, b, p, %, /s, /m, /h, or /t, case insensitive and permitted when appropriate (e.g. W only for power; no b on fluids; p only if Project.current.preferences.fluidUnit has a value).
+        /// </summary>
+        /// <param name="str">The string to parse.</param>
+        /// <param name="amount">The parsed amount, or an unspecified value if the string could not be parsed.</param>
+        /// <param name="unit">The unit that applies to this value. <see cref="UnitOfMeasure.Celsius"/> is not supported.</param>
+        /// <returns>True if the string could be parsed as the specified unit, false otherwise.</returns>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="unit"/> is <see cref="UnitOfMeasure.Celsius"/>.</exception>
         public static bool TryParseAmount(string str, out float amount, UnitOfMeasure unit) {
+            if (unit is UnitOfMeasure.Celsius) { throw new ArgumentException("Parsing to UnitOfMeasure.Celcius is not supported.", nameof(unit)); }
+
             var (mul, _) = Project.current.ResolveUnitOfMeasure(unit);
-            int lastValidChar = 0;
-            float multiplier = unit == UnitOfMeasure.Megawatt ? 1e6f : 1f;
+            float multiplier = unit is UnitOfMeasure.Megawatt or UnitOfMeasure.Megajoule ? 1e6f : 1f;
+
+            str = str.Replace(" ", ""); // Remove spaces to support parsing from the "10 000" precise format, and to simplify the regex.
+            var groups = ParseAmountRegex().Match(str).Groups;
             amount = 0;
-            foreach (char c in str) {
-                if (c is (>= '0' and <= '9') or '.' or '-' or 'e') {
-                    ++lastValidChar;
-                }
-                else {
-                    if (lastValidChar == 0) {
-                        return false;
-                    }
+            if (groups.Count < 4 || !float.TryParse(groups[1].Value, out amount)) {
+                return false;
+            }
+
+            switch (groups[2].Value.SingleOrDefault()) { // μukMGT
+                case 'u' or 'μ':
+                    multiplier = 1e-6f;
+                    break;
+                case 'k' or 'K':
+                    multiplier = 1e3f;
+                    break;
+                case 'm' or 'M':
+                    multiplier = 1e6f;
+                    break;
+                case 'g' or 'G':
+                    multiplier = 1e9f;
+                    break;
+                case 't' or 'T':
+                    multiplier = 1e12f;
+                    break;
+                case 'U' or 'Μ' or 'K': // capital u or μ, or Kelvin symbol; false positive in the regex match
```

Comment:
`M` and `K` are already  matched in earlier `case` statements?

Reply:
Those are the capital mu (U+039C) and the Kelvin symbol (U+212A); the case insensitive regex match on μ and k matches those too. (It appears the Kelvin symbol has a lowercase form for compatibility reasons.)

I guess I decided that having three false positives in the match was a better choice than having to double all the characters `@"^([-+0-9.eE]+)([μukKmMgGtT]?)(/[hHmMsStT]|[WwJjsSbBpP%]?)$"`, and remember to double any new ones, but I didn't think about that too hard.

Reply:
Oh... they really look like a regular M and K... :smile: